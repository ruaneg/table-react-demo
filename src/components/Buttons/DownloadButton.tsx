import { FaDownload } from "react-icons/fa";
import styled from "styled-components";
import { IconContainer } from "../Icons/Icons.styles";

const StyledDownloadButton = styled.button`
  border: 0;
  background: none;
  box-shadow: none;
  border-radius: 0px;
  font-size: 1rem;
  color: #000000d6;
  cursor: pointer;
  &:active {
    color: dodgerblue;
  }
  &:disabled {
    pointer-events: none;
    opacity: 0.6;
  }
`;

const DownloadButton = ({
  handleOnClick,
  disabled,
  label,
  ...rest
}: {
  handleOnClick: () => void;
  disabled: boolean;
  label: string;
}) => {
  return (
    <StyledDownloadButton
      aria-label={"Download Selected files"}
      onClick={handleOnClick}
      disabled={disabled}
      {...rest}
    >
      <IconContainer>
        <FaDownload />
      </IconContainer>
      {label}
    </StyledDownloadButton>
  );
};

export default DownloadButton;
