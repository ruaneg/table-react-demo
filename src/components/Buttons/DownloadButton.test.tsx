import { render, fireEvent } from "@testing-library/react";
import "jest-styled-components";
import { describe, test, expect, vi } from "vitest";
import DownloadButton from "./DownloadButton";

describe("DownloadButton", () => {
  test("renders correctly", () => {
    const { container } = render(
      <DownloadButton
        handleOnClick={() => {}}
        label="Download"
        disabled={false}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("calls handleOnClick when clicked", () => {
    const handleOnClick = vi.fn();
    const { getByLabelText } = render(
      <DownloadButton
        handleOnClick={handleOnClick}
        label="Download"
        disabled={false}
      />
    );
    const button = getByLabelText("Download Selected files");
    fireEvent.click(button);
    expect(handleOnClick).toHaveBeenCalledTimes(1);
  });

  test("applies correct styles when active", () => {
    const { getByLabelText } = render(
      <DownloadButton
        handleOnClick={() => {}}
        label="Download"
        disabled={false}
      />
    );
    const button = getByLabelText("Download Selected files");

    // Simulate active state
    fireEvent.mouseDown(button);

    expect(button).toHaveStyleRule("color", "dodgerblue", {
      modifier: ":active",
    });
  });

  test("applies correct styles when disabled", () => {
    const { getByLabelText } = render(
      <DownloadButton handleOnClick={() => {}} label="Download" disabled />
    );
    const button = getByLabelText("Download Selected files");

    expect(button).toHaveStyleRule("pointer-events", "none", {
      modifier: ":disabled",
    });
    expect(button).toHaveStyleRule("opacity", "0.6", {
      modifier: ":disabled",
    });

    // Ensure click is not triggered
    fireEvent.click(button);
    expect(button).toBeDisabled();
  });
});
