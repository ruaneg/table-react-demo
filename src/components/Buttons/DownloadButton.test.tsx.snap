// Vitest Snapshot v1

exports[`DownloadButton > renders correctly 1`] = `
.c1 {
  padding: 0 8px;
  font-size: 0.875rem;
}

.c0 {
  border: 0;
  background: none;
  box-shadow: none;
  border-radius: 0px;
  font-size: 1rem;
  color: #000000d6;
  cursor: pointer;
}

.c0:active {
  color: dodgerblue;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

<button
  aria-label="Download Selected files"
  class="c0"
>
  <span
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
      />
    </svg>
  </span>
  Download
</button>
`;
